tiles.lua

```diff
@@ -143,11 +167,41 @@ local _checkProperty = {
 		return value == nil or type(value) == "function"
 	end,
 }
+
+--- Contains functions for type-checking arguments.
+-- If a function expects an argument of type `typename', then
+-- `checkProperty[typename](val)` returns true iff `val' is of type
+-- `typename'.
+-- @field name A Lua string or nil.
+-- @field number A Lua number.
+-- @field positive_number A _non-negative_ Lua number.
+-- @fixme positive_number lies about its name!
+-- @field string A Lua string.
+-- @field boolean A Lua boolean.
+-- @field alignment A valid pair of alignment specifiers.
+-- @field userdata Any value.
+-- @field function A Lua function.
+-- @field percent A Lua number between 0 and 1 (inclusive).
+-- @field alpha An integer between 0 and 255 (inclusive).
+-- @field colour An integer between 0x000000 and 0xFFFFFF (inclusive).
+-- @fixme alias `color' for 'muricans.
+-- @field gradient An integer equal to either 1 or 2.
+-- @field simplePoint A table which has fields `x' and `y' with values which are Lua numbers.
+-- @field complexPoint A `simplePoint' which optionally contains the fields `rgb' or `opacity' with values which are a `colour' and a `percent' respectively.
+-- @field simplePoints An array of `simplePoint' values.
+-- @field complexPoints An array of `complexPoint' values.
+-- @field meta An integer between 0 and 15 (inclusive).  Represents a Minecraft item meta value.
+-- @field surface A `surface' object as exposed by the Terminal Glasses API.
+-- @field capture A `capture' object as exposed by the Terminal Glasses API.
+-- @field bridge A wrapped Terminal Glasses Bridge (for example, the object returned by the call peripheral.wrap(sideOfBridge)).
+-- @field functionOrNil A Lua function or nil.
 checkProperty = {}
 for property, checker in pairs(_checkProperty) do
 	checkProperty[property] = checker
 end
 
+--- No idea.
+-- @todo What exactly do these do...?
```

Comment:
These translate the arguments the user passes into the API into arguments the API is expecting.

For example, with alignment it gets passed as object:SetObjectAlignment(horizontal, vertical), but the API stores it as a table {horizontal, vertical}, so this function does the neccessary conversion.

For the points, it's simply to strip away any additional indexes from the tables passed in by the user. It also means a completely new copy is made just in case the user modifies the table after passing it into the API.

The formatPropertyOut functions do the same, just in reverse.